app/scripts/metamask-controller.js

```diff
@@ -688,6 +689,8 @@ export default class MetamaskController extends EventEmitter {
           `${this.approvalController.name}:hasRequest`,
           `${this.approvalController.name}:acceptRequest`,
           `${this.approvalController.name}:rejectRequest`,
+          `SnapController:getPermitted`,
+          `SnapController:install`,
```

Comment:
why not `${this.snapController.name}:getPermitted` and ${this.snapController.name}:install`?

Reply:
No reason, we can do that 👍

Reply:
because `${this.snapController.name}` is undefined

Reply:
Oh yeah, we can't since the PermissionController is defined before the SnapController

Reply:
We could move it around, but I don't think that its worth the hassle right now